fullstory-segment-middleware/src/test/java/com/fullstorydev/fullstory_segment_middleware/FullStorySegmentMiddlewareTest.java

```diff
@@ -0,0 +1,443 @@
+package com.fullstorydev.fullstory_segment_middleware;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import com.fullstory.FS;
+import com.segment.analytics.Middleware;
+import com.segment.analytics.integrations.AliasPayload;
+import com.segment.analytics.integrations.BasePayload;
+import com.segment.analytics.integrations.GroupPayload;
+import com.segment.analytics.integrations.IdentifyPayload;
+import com.segment.analytics.integrations.ScreenPayload;
+import com.segment.analytics.integrations.TrackPayload;
+import com.segment.analytics.Properties;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.internal.verification.VerificationModeFactory;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({FS.class, Log.class})
+public class FullStorySegmentMiddlewareTest {
+
+    @Mock
+    Context mockContext;
+    @Mock
+    SharedPreferences mockPrefs;
+    @Mock
+    Middleware.Chain mockChain;
+
+    FullStorySegmentMiddleware fullStorySegmentMiddleware;
+
+    @Before
+    public void setUp(){
+        MockitoAnnotations.initMocks(this);
+        when(mockContext.getSharedPreferences(anyString(), anyInt())).thenReturn(mockPrefs);
+
+        PowerMockito.mockStatic(FS.class);
+        when(FS.getCurrentSessionURL(anyBoolean())).thenReturn("mock FullStory Session URL");
+        when(FS.getCurrentSessionURL()).thenReturn("mock FullStory Session URL");
+    }
+
+    @Test
+    public void constructor_WithAllowListedEvent() {
+        ArrayList<String> allowList = new ArrayList<>();
```

Comment:
we should change the type to `List` so we are also testing that the `FullStorySegmentMiddleware` constructor takes in the lowest type